synapse/lib/agenda.py

```diff
@@ -849,15 +856,19 @@ async def _runJob(self, user, appt):
 
                 took = finishtime - starttime
                 mesg = f'Agenda completed query for iden={appt.iden} name={appt.name} with result "{result}" ' \
-                       f'took {took:.3f}s'
+                       f'took {took:.3f}s.'
+                if not self.core.isactive:
+                    mesg = mesg + ' Agenda status will not be saved since the Cortex is no longer active.'
                 logger.info(mesg, extra={'synapse': {'iden': appt.iden, 'name': appt.name, 'user': user.iden,
                                                      'result': result, 'username': user.name, 'took': took}})
                 edits = {
                     'lastfinishtime': finishtime,
                     'isrunning': False,
                     'lastresult': result,
                 }
-                await self.core.addCronEdits(appt.iden, edits)
+
+                if self.core.isactive:
```

Comment:
If this ( and the above isactive check ) are the meat of the changes, they LGTM 👍